### PR TITLE
	new file:   rules/windows/process_creation/process_creation_win_msdt…

### PR DESCRIPTION
**Pull request by greg-workspace**
Hello, I add one more rule to detect CVE-2022-30190: sdiagnhost.exe loaded with powershell module
…_sdiagnhost.yml

### CODE REVIEW

File: rules/windows/process_creation/process_creation_win_msdt_sdiagnhost.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  selection_cmd:
+    CommandLine|contains:
+      - 'PCWDiagnostic'
+      - '/af ' # For answer files
+  selection_load:
+    Image|endswith: '\sdiagnhost.exe'
+    ImageLoaded|contains: 'PowerShell.Command'
+    ImageLoaded|endswith: '.dll'
+
+  condition: all of selection*
```
Comment (Neo23x0) [inline rules/windows/process_creation/process_creation_win_msdt_sdiagnhost.yml:28]: How can `Image` simultaneously end with `\msdt.exe` AND `\sdiagnhost.exe` ?

Did you mean `ParentImage` for `msdt.exe`?

**Comment by greg-workspace** [comment]
The process tree is explained as the attachment.

And who is the parent process of "sdiagnhost.exe"?

I need to do more reversing to find the parent process , do you have any
suggestions  about the rule modification?

Florian Roth ***@***.***> 於 2022年6月10日 週五 下午6:46寫道：

**Comment by phantinuss** [comment]
@eiger: The ImageLoaded field is available for e.g. Sysmon Event ID 7 and is mapped in Sigma as `category: image_load`. I would suggest to create a separate rule for each category as Sigma rules are sorted by their category.
 
 The selection via img/cmd seems to be copied from rule 258fc8ce-8352-443a-9120-8a11e4857fa5 and therefore is redundant. Also the condition won't work as is, see @Neo23x0's remark

**Comment by greg-workspace** [comment]
Instead of detecting process creation: sdiagnhost.exe, detecting msdt.exe loading sdiageng.dll is more precise.

**Comment by greg-workspace** [comment]
> @eiger: The ImageLoaded field is available for e.g. Sysmon Event ID 7 and is mapped in Sigma as `category: image_load`. I would suggest to create a separate rule for each category as Sigma rules are sorted by their category. See the rules/windows/image_load folder for examples.
> 
> The selection via img/cmd seems to be copied from rule 258fc8ce-8352-443a-9120-8a11e4857fa5 and therefore is redundant. Also the condition won't work as is (see @Neo23x0's remark).

Thank you for your suggestion. I create a separate the rule which moving from process_creation to image_load.

**Pull request closed without merging**